service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -108,13 +108,22 @@ private RunSetDetailsResponse convertToRunSetDetails(RunSet runSet) {
 
   @Override
   public ResponseEntity<RunSetListResponse> getRunSets(UUID methodId, Integer pageSize) {
-    List<RunSet> runSets = runSetDao.getRunSets();
-    List<RunSet> updatedRunSets = smartRunSetsPoller.updateRunSets(runSets);
-
-    List<RunSetDetailsResponse> runSetDetails =
-        updatedRunSets.stream().map(this::convertToRunSetDetails).toList();
-    RunSetListResponse response = new RunSetListResponse().runSets(runSetDetails);
-
+    List<RunSet> updatedRunSets;
+    RunSetListResponse response;
+
+    if (methodId != null) {
+      List<RunSet> filteredRunSet = Collections.singletonList(runSetDao.getRunSet(methodId));
+      updatedRunSets = smartRunSetsPoller.updateRunSets(filteredRunSet);
+      List<RunSetDetailsResponse> filteredRunSetDetails =
+          updatedRunSets.stream().map(this::convertToRunSetDetails).toList();
+      response = new RunSetListResponse().runSets(filteredRunSetDetails);
```

Comment:
These three lines look(?) mostly the same to me... I think you could put only the choice of whether to use `getRunSet(methodId)` or `runSetDao.getRunSets()` in the `if` and keep the rest as a shared code path?